### PR TITLE
Documentation about mail integrations with SMTP relays

### PR DESCRIPTION
**Pull request by NyanKiyoshi**
This PR introduce a few new sections to help the user to setup the email settings in Saleor, especially for setting up SMTP relays (sendgrid, mailgun and mailjet).

---

It also add a fix for #2088. Which, instead of setting the `DEFAULT_FROM_EMAIL` value to `None`, it sets the value to the email username, which means, for example, in the case of gmail: hello@gmail.com. 

The reason is quite simple: in the email url we have to set the username to an address mail for SMTP servers (not for every relays though).

e.g.: `smtp://my.gmail.username:my-password@smtp.gmail.com:465/?ssl=True`

And, in the case the user did not set any email, it won't fail, as the target is the console and not a SMTP server (or relay server).

The documentation is also now saying in relays that we need to set this value for some relays.

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] Python code quality checks pass: `pycodestyle`, `pydocstyle`, `pylint`.
1. [x] JavaScript code quality checks pass: `eslint`.

### CODE REVIEW

**Comment by NyanKiyoshi** [comment]
Hi,

I will add SES (as SMTP) as well as it may really be useful for users for their deployment.

And some more information about how to customize the sender information and subject prefix.

**Comment by joshkpeterson** [comment]
Setting up Saelor for the first time. Without the fix here, my installation on heroku was crashing. Kind of alarming such a small thing could bring down the app. Would be great to get this merged in as this was a distraction for me.

**Comment by NyanKiyoshi** [comment]
Did someone had the opportunity to look over this pull request?

File: docs/guides/email_integration.rst [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+You can set the environment variable ``EMAIL_URL`` to the SMTP URL,
+which will contain a straightforward value as shown in below examples.
+
+
+.. table::
+
+   ==========================================================  ================================================================================
+   Description                                                 URL
+   ==========================================================  ================================================================================
+   GMail with **SSL on**.                                      ``smtp://my.gmaail.username@gmail.com:my-password@smtp.gmail.com:465/?ssl=True``
```
Comment (patrys) [inline docs/guides/email_integration.rst]: It says `gmaail`. Typo?

**Pull request merged**